toolkits/microsoft/arcade_microsoft/outlook_calendar/tools/create_event.py

```diff
@@ -0,0 +1,68 @@
+from typing import Annotated
+
+from arcade.sdk import ToolContext, tool
+from arcade.sdk.auth import Microsoft
+
+from arcade_microsoft.client import get_client
+from arcade_microsoft.outlook_calendar._utils import (
+    create_timezone_request_config,
+    get_default_calendar_timezone,
+    remove_timezone_offset,
+)
+from arcade_microsoft.outlook_calendar.models import (
+    Attendee,
+    DateTimeTimeZone,
+    Event,
+)
+
+
+@tool(requires_auth=Microsoft(scopes=["MailboxSettings.Read", "Calendars.ReadWrite"]))
+async def create_event(
+    context: ToolContext,
+    subject: Annotated[str, "The text of the event's subject (title) line."],
+    body: Annotated[str, "The body of the event"],
+    start_date_time: Annotated[
+        str,
+        "The datetime of the event's start, represented in "
+        "ISO 8601 format. Timezone offset is ignored. For example, 2025-04-25T13:00:00",
+    ],
+    end_date_time: Annotated[
+        str,
+        "The datetime of the event's end, represented in "
+        "ISO 8601 format. Timezone offset is ignored. For example, 2025-04-25T13:30:00",
+    ],
+    location: Annotated[str | None, "The location of the event"] = None,
+    attendee_emails: Annotated[
+        list[str] | None,
+        "The email addresses of the attendees of the event. "
+        "Must be valid email addresses e.g., username@domain.com.",
```

Comment:
perhaps add an email validation step before trying to create the event with this arg values?

Reply:
good point. I had previously assumed that microsoft api would return a helpful error in this case, but turns out it doesn't